File: networklib/src/main/java/com/brins/networklib/model/personal/PersonalizedMusicList.kt
package com.brins.networklib.model.personal

import com.brins.baselib.module.BaseData
import com.brins.baselib.module.ITEM_HOME_PERSONALIZED

/**
 * @author lipeilin
 * @date 2020/7/22
 */
class PersonalizedMusicList : BaseData() {

    var id = ""

    var type = 0

    var name = ""

    var copywriter = ""

    var picUrl = ""

    var canDislike = false

    var trackNumberUpdateTime: Long = 0

    var playCount = 0

    var trackCount = 0

    var highQuality = false

    var alg = ""
    override val itemType: Int
        get() = ITEM_HOME_PERSONALIZED


}

File: lib/models/personalizedmusiclist/personalized_music_list.dart
import 'package:json_annotation/json_annotation.dart';

part 'personalized_music_list.g.dart';

@JsonSerializable()
class PersonalizedMusicList {
  int id;

  int type = 0;

  String name;

  String copywriter;

  String picUrl;

  bool canDisLike = false;

  int trackNumberUpdateTime;

  int playCount = 0;

  int trackCount = 0;

  bool highQuality = false;

  String alg;

  PersonalizedMusicList(
      this.alg,
      this.canDisLike,
      this.copywriter,
      this.highQuality,
      this.id,
      this.name,
      this.picUrl,
      this.playCount,
      this.trackCount,
      this.trackNumberUpdateTime,
      this.type);

  factory PersonalizedMusicList.fromJson(Map<String, dynamic> json) =>
      _$PersonalizedMusicListFromJson(json);

  Map<String, dynamic> toJson() => _$PersonalizedMusicListToJson(this);
}
